### PR TITLE
Combining RTMPErrors and ChangeTrackMuteRequests

### LINKED ISSUE #115: Events are using different MutableSharedFlows.
In the MeetingViewModel, there are two events. One for `rtmpErrors` and another for `changeTrackMuteRequest`.
They should be moved into a single Flow with a shared sealed data class structure.

For an explanation of what this is, look [here](https://www.raywenderlich.com/22030171-reactive-streams-on-kotlin-sharedflow-and-stateflow).

### PR DESCRIPTION
**Pull request by Stuie**
Fixes #115

RTMPError and ChangeTrackMuteRequests are now emitted via the same Flow, using a sealed Event class. The collection in MeetingFragment.kt has been updated accordingly.

This is intended as a preliminary PR. I have not tested the change yet, but before I spend a long time on testing, I wanted to make sure I had the right idea.

I also wanted to get some opinions on naming, as I suspect that `Event` is a little bit too generic. :)

### CODE REVIEW

**Comment by AniketSK** [comment]
Thanks so much @Stuie, this is exactly what I'd imagined it should be. The naming is fine. I'll run a few tests on this myself and get back in a bit.

File: app/src/main/java/live/hms/app2/ui/meeting/MeetingFragment.kt [hunk 1]
```diff
@@ -411,25 +411,20 @@ class MeetingFragment : Fragment() {
       }
     }
 
-    viewLifecycleOwner.lifecycleScope.launch {
-      meetingViewModel.rtmpErrors.collect { rtmpException ->
-        if (rtmpException != null) {
-          withContext(Dispatchers.Main) {
-            Toast.makeText(context, "RTMP error $rtmpException", Toast.LENGTH_LONG).show()
+    lifecycleScope.launch {
```
Comment (AniketSK) [inline app/src/main/java/live/hms/app2/ui/meeting/MeetingFragment.kt]: The entire lifecycle of the fragment extends far beyond the view and I'd want this to be stopped as soon as the view is destroyed so it's important that the operations be carried out on the viewLifeCycleOwner.

```suggestion
    viewLifecycleOwner.lifecycleScope.launch {
```

**Review by AniketSK — CHANGES_REQUESTED** [review]
Just the one change I suggested and opening the PR to dev instead of release-v2 and this should good! Thanks again @Stuie

**Review by AniketSK — APPROVED** [review]

**Comment by Stuie** [comment]
Hey, sorry for not following up. I took a quick look last night, but I'm moving house right now and didn't have my dev machine set up to make the change you proposed.

**Comment by AniketSK** [comment]
@Stuie

> Hey, sorry for not following up. I took a quick look last night, but I'm moving house right now and didn't have my dev machine set up to make the change you proposed.

Don't worry about it, the changes required were minor and I just made them. Thanks for contributing! Much appreciated.

**Pull request merged**